File: CLI with SQL Server/src/apps/console/BasketballConsole/BasketballConsole/Repo/Basketball.cs
using NBA.Interfaces;

namespace NBA.Repo
{
    internal static class Basketball
    {
        private static IBasketballRepo? repo;

        public static IBasketballRepo Repo { get => repo; private set => repo = value; }

        public static void SetRepo(IBasketballRepo repo)
        {
            Repo = repo;
        }
    }
}

File: Java with MongoDB/server/Basketball/Microservice/basketball-microservice/src/main/java/com/nba/basketball_microservice/infrastructure/Basketball.java
package com.nba.basketball_microservice.infrastructure;

public class Basketball {
    private static IBasketballRepo repo;

    public static IBasketballRepo getRepo() {
        return repo;
    }

    public static void setRepo(IBasketballRepo repo) {
        Basketball.repo = repo;
    }
}
